Copy types.d.ts into dist folder for packaging

creedasaurus opened the pull request:
Right now the npm package for this doesn't include the `types.d.ts` file which is too bad when you're writing typescript and enjoy the benefits of typechecking. I added this to simply copy the types file into the dist folder which is included in the npm bundle. I tested it locally with `npm link` and it worked great.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

szechyjs commented:
Can we merge this finally?

dchauhan-twilio reviewed (commented):
Firstly apologies on not addressing this PR sooner.  Overall the change looks good, do you mind rebasing/merging the latest to resolve the conflict. Happy to approve afterwards.

creedasaurus commented:
K. I updated it. I also fixed the copy plugin version to the version of webpack you all are using. 
I also fixed the build since there were a few comments that were tripping the `eslint` check. Hope that's ok.

Took me a minute to get my node to build this for some reason. Something weird with older dependency versions or something?

creedasaurus commented:
Any reason to keep this build setup with such a dated version of Node (`>=6` or `8` for your build system)? If not, my inner-urge to just do a quick update to a more recent LTS could open another PR, hahaha.

creedasaurus commented:
Woops. Sorry. Accidentally hit the close button on GitHub mobile.

dchauhan-twilio commented:
Thanks for updating the PR. It has also picked up the workflows, and they are passing too.

Another PR just got merged before yours. Do you mind resolving the conflicts again with a merge or rebase?
